Weather Man/weather_parser.py

```diff
@@ -0,0 +1,111 @@
+"""
+weather_parser.py
+
+This module defines the WeatherParser class, which parses weather data files year-wise and month-wise for Lahore,
+Pakistan.
+
+Classes:
+- WeatherParser: A class to parse weather data files year-wise and month-wise from a specified directory.
+
+"""
+
+import os
+from constants import month_map
+from weather_data import WeatherData
+import pandas as pd
+from datetime import datetime
+
+
+class WeatherParser:
+    """
+        A class to parse weather data files year-wise and month-wise from a specified directory.
+
+        Attributes:
+        - files_dir (str): Directory path where weather data files are located.
+
+        Methods:
+        - parse_files_year_wise(year): Parses all weather data files for a specific year.
+        - parse_files_month_wise(year, month): Parses all weather data files for a specific month in a year.
+        """
+
+    def __init__(self, files_dir):
+        """
+        Initializes WeatherParser with the directory containing weather data files.
+
+        Args:
+        - files_dir (str): Directory path where weather data files are located.
+        """
+
+        self.files_dir = files_dir
+
+    def parse_file(self, file_path):
+        """
+        Parses a CSV file containing weather data and returns a WeatherData object.
+
+        Args:
+        - file_path (str): Path to the CSV file containing weather data.
+
+        Returns:
+        - Object: A WeatherData object containing parsed weather readings.
+        """
+        readings = []
+        data = pd.read_csv(file_path)
+        data.columns = data.columns.str.strip()
+        date_col = None
+        if "PKT" in data.columns:
+            date_col = "PKT"
+        elif "PKST" in data.columns:
+            date_col = "PKST"
+        data = data[
+            [date_col, "Max TemperatureC", "Mean TemperatureC", "Min TemperatureC", "Max Humidity", "Mean Humidity",
+             "Min Humidity"]]
+        data.dropna(inplace=True)
+        for _, row in data.iterrows():
+            date = datetime.strptime(row[f"{date_col}"], "%Y-%m-%d")
+            max_temp = row["Max TemperatureC"]
+            mean_temp = row["Mean TemperatureC"]
+            min_temp = row["Min TemperatureC"]
+            max_humidity = row["Max Humidity"]
+            mean_humidity = row["Mean Humidity"]
+            min_humidity = row["Min Humidity"]
+            reading = WeatherData(date, max_temp, mean_temp, min_temp, max_humidity, mean_humidity, min_humidity)
+            readings.append(reading)
```

Comment:
validation not on the data, Also no need to have variable here as you are just passing them to the Class